pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,533 @@
+
+	if err := r2.Err(); err != nil {
+		return nil, err
+	}
+
+	err = r2.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		pod, ok := info.Object.(*corev1.Pod)
+		if !ok {
+			return fmt.Errorf("expected *corev1.Pod, got %T", info.Object)
+		}
+
+		podInfo := struct {
+			Name     string
+			BatchID  string
+			Status   string
+			Ready    string
+			Age      string
+			Restarts string
+			Revision string
+		}{
+			Name:     pod.Name,
+			BatchID:  pod.Labels["rollouts.kruise.io/rollout-batch-id"],
+			Status:   string(pod.Status.Phase),
+			Age:      duration.HumanDuration(time.Since(pod.CreationTimestamp.Time)),
+			Restarts: fmt.Sprintf("%v (%v ago)", strconv.Itoa(int(pod.Status.ContainerStatuses[0].RestartCount)), duration.HumanDuration(time.Since(pod.Status.ContainerStatuses[0].LastTerminationState.Terminated.FinishedAt.Time))),
+		}
+
+		// Calculate ready status
+		readyContainers := 0
+		for _, containerStatus := range pod.Status.ContainerStatuses {
+			if containerStatus.Ready {
+				readyContainers++
+			}
+		}
+		podInfo.Ready = fmt.Sprintf("%d/%d", readyContainers, len(pod.Spec.Containers))
+
+		// Calculate revision
+		if pod.Labels["pod-template-hash"] != "" {
+			podInfo.Revision = pod.Labels["pod-template-hash"]
+		} else if pod.Labels["controller-revision-hash"] != "" {
+			podInfo.Revision = pod.Labels["controller-revision-hash"]
+		}
+
+		workloadInfo.Pod = append(workloadInfo.Pod, podInfo)
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return workloadInfo, nil
+}
+
+func (o *DescribeRolloutOptions) colorizeIcon(phase string) string {
+	if o.NoColor || phase == "" {
+		return ""
+	}
+	switch phase {
+	case string(rolloutsapi.RolloutPhaseHealthy):
+		return "\033[32m✔\033[0m"
+	case string(rolloutsapi.RolloutPhaseProgressing):
+		return "\033[33m⚠\033[0m"
+	case string(rolloutsapi.RolloutPhaseDisabled), string(rolloutsapi.RolloutPhaseTerminating), string(rolloutsapi.RolloutPhaseDisabling):
+		return "\033[31m✘\033[0m"
+	case string(rolloutsapi.RolloutPhaseInitial):
+		return "\033[33m⚠\033[0m"
+	default:
+		return ""
+	}
+}
+
+func (o *DescribeRolloutOptions) printRolloutInfo(rollout *rolloutsapi.Rollout) {
+	// Name, Namespace, Status, Message
+	fmt.Fprintf(o.Out, tableFormat, "Name:", rollout.Name)
+	fmt.Fprintf(o.Out, tableFormat, "Namespace:", rollout.Namespace)
+	if rollout.Status.ObservedGeneration == rollout.GetObjectMeta().GetGeneration() {
+		fmt.Fprintf(o.Out, tableFormat, "Status:", o.colorizeIcon(string(rollout.Status.Phase))+" "+string(rollout.Status.Phase))
+
+		if rollout.Status.Message != "" {
+			fmt.Fprintf(o.Out, tableFormat, "Message:", rollout.Status.Message)
+		}
+	}
+
+	// Strategy
+	fmt.Fprintf(o.Out, tableFormat, "Strategy:", "Canary")
+
+	if canary := rollout.Spec.Strategy.Canary; canary != nil {
+		fmt.Fprintf(o.Out, tableFormat, " Step:", strconv.Itoa(len(canary.Steps))+"/"+strconv.Itoa(int(rollout.Status.CanaryStatus.CurrentStepIndex)))
```

Comment:
plz switch strconv.Itoa(len(canary.Steps)) and strconv.Itoa(int(rollout.Status.CanaryStatus.CurrentStepIndex) in the step